### PR TITLE
Add DRAM expansion launch warning

### PR DESCRIPTION
**Pull request by MutantAura**
Self-explanatory. Setting currently causes some titles to crash on launch, causing a lot of un-needed support requests.
Can potentially be reverted when #3314 is resolved.
image

### CODE REVIEW

**Comment by MetrosexualGarbodor** [comment]
IMO, people will simply ignore the warning, so it would be best to either detect the game IDs (Pokémon Scarlet/Violet and Arceus) and automatically disable the option, or if we don't want to do game IDs, disable it when no mods are loaded. 

If neither of those are possible this is good too, just saying there's a high probability of people skipping it.

**Comment by MutantAura** [comment]
> IMO, people will simply ignore the warning, so it would be best to either detect the game IDs (Pokémon Scarlet/Violet and Arceus) and automatically disable the option, or if we don't want to do game IDs, disable it when no mods are loaded.
> 
> If neither of those are possible this is good too, just saying there's a high probability of people skipping it.

The aim here, even if they skip it, is to trigger some neurons. If they receive a warning, ignore it and then their game crashes, they're far more likely to actually read it the next time. Which is why I haven't decided to allow the option selection to be saved.

**Comment by WiseHorror** [comment]
My only suggestion with this warning would be to also mention that this setting does not impact performance.

**Comment by EmulationFanatic** [comment]
This approach seems a bit like using a sledgehammer to drive a pushpin. How many games are known to crash or have instability with the DRAM expansion enabled?  The most obvious is Pokemon Legends Arceus, so would it be possible to start with the pop-up for only that title ID?

I'll separately review with my verbiage suggestions :)

File: Ryujinx.Ava/Assets/Locales/en_US.json [hunk 1]
```diff
@@ -594,5 +594,7 @@
   "SettingsTabHotkeysVolumeDownHotkey": "Decrease Volume:",
   "VolumeShort": "Vol",
   "SettingsEnableMacroHLE": "Enable Macro HLE",
-  "SettingsEnableMacroHLETooltip": "High-level emulation of GPU Macro code.\n\nImproves performance, but may cause graphical glitches in some games.\n\nLeave ON if unsure."
+  "SettingsEnableMacroHLETooltip": "High-level emulation of GPU Macro code.\n\nImproves performance, but may cause graphical glitches in some games.\n\nLeave ON if unsure.",
+  "DialogPerformanceCheckExpandDramEnabledMessage": "You have DRAM expansion enabled, which is designed to be used for 4K game-mods ONLY.",
+  "DialogPerformanceCheckExpandDramEnabledConfirmMessage": "This setting is known to cause crashes and instability. If you are unsure if you need this, then you don't. Would you like to disable DRAM expansion now?"
```
Comment (EmulationFanatic) [inline Ryujinx.Ava/Assets/Locales/en_US.json:599]: Suggestion: "This setting, when enabled, is currently known to cause crashes and instability in some games including Pokemon Legends Arceus. If you are unsure whether you need this option, **leave it disabled**. Disable DRAM expansion now?"

File: Ryujinx/Ui/MainWindow.cs [hunk 2]
```diff
@@ -683,6 +683,24 @@ public void PerformanceCheck()
...
                 shadersDumpWarningDialog.Dispose();
             }
+
+            if (ConfigurationState.Instance.System.ExpandRam.Value)
+            {
+                MessageDialog dramWarningDialog = new MessageDialog(this, DialogFlags.Modal, MessageType.Warning, ButtonsType.YesNo, null)
+                {
+                    Title         = "Ryujinx - Warning",
+                    Text          = "You have DRAM expansion enabled, which is designed to be used for 4K game-mods ONLY.",
+                    SecondaryText = "This setting is known to cause crashes and instability. If you are unsure if you need this, then you don't. Would you like to disable DRAM expansion now?"
```
Comment (EmulationFanatic) [inline Ryujinx/Ui/MainWindow.cs:693]: Suggestion: "This setting, when enabled, is currently known to cause crashes and instability in some games including Pokemon Legends Arceus. If you are unsure whether you need this option, **leave it disabled**. Disable DRAM expansion now?"

**Review by EmulationFanatic — COMMENTED** [review]
Added some verbiage suggestions for clarity/conciseness.

**Comment by Serfrost** [comment]
@EmulationFanatic Due to recent circumstances, I've seen more crashes with this setting with Scarlet \ Violet. I'd simply encourage not naming any specific games unless the dialogue changes automatically based on the title ID being launched.

On the contrary I don't really think an entire pop-up is necessary, rather, turning off the option as an override for some games is more or less going to be the least enduser-involved choice.

However, preventing whatever is causing the crashes outright with a patch for those specific titles may be the most sensible option (to avoid the ire of mod devs,) even if it isn't within spec.

**Comment by MutantAura** [comment]
If the user selects "No" to disabling the expansion then the prompt will now not appear again. I've also taken the liberty of resetting the Expand RAM setting in general as well. We're seeing far too many people with it active. 🗡️

File: Ryujinx.Ava/Assets/Locales/en_US.json (hunk 1 above)
Comment (MutantAura) [reply Ryujinx.Ava/Assets/Locales/en_US.json:599]: Not a huge fan of mentioning specific titles, especially when that crash is technically a regression. I'd rather leave it vague. Perhaps just: `This setting, when enabled, is currently known to cause crashes and instability in some games. If you are unsure whether you need this option, leave it disabled. Disable DRAM expansion now?".

Comment (EmulationFanatic) [reply Ryujinx.Ava/Assets/Locales/en_US.json:599]: Sure.

**Comment by Serfrost** [comment]
> @MutantAura I've also taken the liberty of resetting the Expand RAM setting in general as well. We're seeing far too many people with it active.

Wholeheartedly agree with this decision, otherwise those who currently have it enabled would likely outweigh those who decide to read it and turn it off willingly after the change.

**Comment by gdkchan** [comment]
Is that still necessary now that #3870 was merged?

**Comment by MutantAura** [comment]
Possibly not. I wanted to give it a few days to see if #3870 was having the desired effect. 
Happy to close for now and re-assess if we ever hit similar issues in the future.

**Pull request closed without merging**